legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/test/RelationalConnectionFactory.java

```diff
@@ -50,4 +57,25 @@ public Optional<Connection> tryBuildFromConnection(Connection connection, String
     {
         return ConnectionFactoryExtension.super.tryBuildFromConnection(connection, testData, element);
     }
+
+    @Override
+    public Optional<Pair<Connection, List<Closeable>>> tryBuildTestConnection(Connection sourceConnection, EmbeddedData data)
+    {
+        if(data instanceof RelationalCSVData && sourceConnection instanceof RelationalDatabaseConnection)
+        {
+            RelationalDatabaseConnection connection = new RelationalDatabaseConnection();
+            connection.databaseType = DatabaseType.H2;
+            connection.type = DatabaseType.H2;
+            connection.element = sourceConnection.element;
+            connection.authenticationStrategy = new TestDatabaseAuthenticationStrategy();
+            LocalH2DatasourceSpecification localH2DatasourceSpecification = new LocalH2DatasourceSpecification();
+            // TODO generate sql with pure helper function
```

Comment:
Lets boot up H2 here itself

Reply:
Looking at the code for relational, the difference is we already have a dedicated H2LocalServer up. 
See: https://github.com/finos/legend-engine/blob/master/legend-engine-xt-relationalStore-executionPlan-connection/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/H2LocalServer.java
There is no need to boot up a specific special server for relational.